android/src/main/java/com/oney/WebRTCModule/CameraCaptureController.java

```diff
@@ -48,7 +60,132 @@ public CameraCaptureController(Context context, CameraEnumerator cameraEnumerato
         this.constraints = constraints;
     }
 
-    public void switchCamera() {
+    @Nullable
+    @Override
+    public String getDeviceId() {
+        return currentDeviceId;
+    }
+
+    private String findDeviceId(String cameraName) {
+        String[] deviceNames = cameraEnumerator.getDeviceNames();
+        for (int i = 0; i < deviceNames.length; i++) {
+            if (Objects.equals(deviceNames[i], cameraName)) {
+                return String.valueOf(i);
+            }
+        }
+        return null;
+    }
+
+    public void applyConstraints(ReadableMap constraints, Consumer<Exception> onFinishedCallback) {
+        ReadableMap oldConstraints = this.constraints;
+        int oldTargetWidth = this.targetWidth;
+        int oldTargetHeight = this.targetHeight;
+        int oldTargetFps = this.targetFps;
+
+        // Don't save constraints yet, since we may fail to find a fit.
+        Runnable saveConstraints = () -> {
+            this.constraints = constraints;
+            this.targetWidth = constraints.getInt("width");
```

Comment:
Are we guaranteed to have some value for width here? What if it's missing from the passed constraints?

Reply:
`normalizeMediaConstraints` ensures we have a width or height if either one is missing, so I think this should be fine.